### ISSUE TITLE
Segfault during backward when using PReLU

### ISSUE THREAD

**Issue by tommm994** [pytorch/pytorch#10723]
I'm trying to train a network using PReLU module, but I get Segfault during the backward. Here's a piece of code that reproduces the bug:

    gt = torch.rand(2,3,256,256)
    gt = torch.autograd.Variable(gt.cuda(async=True))
    input = torch.rand(2,134,256,256)
    input = torch.autograd.Variable(input.cuda())
    lossL1 = torch.nn.L1Loss()
    lossL1 = lossL1.cuda()

    net = nn.Sequential(nn.PReLU(), nn.Conv2d(134, 3, kernel_size=1, stride=1, bias=False)).cuda()

    output = net(input)

    loss = lossL1(output, gt)
    loss.backward()

In this example, my network just consists in a PReLU followed by a simple convolution. Note that if I switch the order of both modules, the Segfault doesn't occur, so it only bugs when the PReLU is the first layer.

Also note that if I don't use the GPU, the Segfault doesn't occur neither.

Rem: I tried with pytorch versions 0.4.0 and 0.4.1.

**Comment by tommm994** [comment]
Here's the thread on the pytorch forum corresponding to this issue:
[https://discuss.pytorch.org/t/segfault-during-backward-when-using-prelu/23574/2](url)

**Comment by tommm994** [comment]
https://discuss.pytorch.org/t/segfault-during-backward-when-using-prelu/23574

**Comment by ptrblck** [comment]
@zou3519 This issue seems to be thrown, if the input does not require gradients:
```python
act = nn.PReLU().to('cuda')
x = torch.randn(1, requires_grad=False, device='cuda')
output = act(x)
output.mean().backward()
print(x.grad)
```
Setting `requires_grad=True` works. 
Maybe it helps debugging this issue.

**Comment by zou3519** [comment]
thanks @ptrblck.

I tried to bisect but this bug has been here for at least 3 months (gulp)

**Comment by ptrblck** [comment]
@zou3519 I could try to help debug it, if you point me into a direction.
